### PR TITLE
Add OBS repository for dovecot without delay

### LINKED ISSUE #72: 500 ms delay between sending a message over SMTP and getting IDLE notified
I have measured round trip time sending messages between two accounts: https://github.com/deltachat/deltachat-core-rust/pull/4974#issuecomment-1806517224

It takes roughly 1.1 seconds to send a message to the echo bot and receive a reply with two Delta Chat accounts. Slightly more than 0.5 seconds (516 or 517 ms) is spent between a mail is sent over SMTP and Dovecot notifies the client in IDLE mode that the message has arrived.

There is a suspicious constant `#define NOTIFY_DELAY_MSECS 500` in Dovecot `mailbox-watch.c`:
https://github.com/dovecot/core/blob/93a53a9d590f0220de28600f36a969cc38c3148b/src/lib-storage/mailbox-watch.c#L12

`mailbox_watch_add` is called from storage implementation. We are using Maildir (may consider switching to dbox, but Maildir is good enough), so in our case it is here:
https://github.com/dovecot/core/blob/93a53a9d590f0220de28600f36a969cc38c3148b/src/lib-storage/index/maildir/maildir-storage.c#L632-L645

Here is a `mailbox_notify_changes` which dynamically dispatches to a particular storage implementation:
https://github.com/dovecot/core/blob/93a53a9d590f0220de28600f36a969cc38c3148b/src/lib-storage/mail-storage.c#L2473-L2484

And finally, here is IMAP IDLE command implementation using `mailbox_notify_changes` to subscribe for notifications:
https://github.com/dovecot/core/blob/93a53a9d590f0220de28600f36a969cc38c3148b/src/imap/cmd-idle.c#L289

So it seems `NOTIFY_DELAY_MSECS 500` directly affects how IDLE callback is called when storage notices changes. And it corresponds with the measurements, it is always slightly more than 500 ms delay.

And here is a commit adding it 14 years ago:
https://github.com/dovecot/core/commit/56fb5d09955b6097f77b341717fd9b70e9f13e7a

It reads:
> mailbox_notify_changes(): Delay sending notifications for 500 msecs.
>
> If the notification is done immediately, IDLE may not notice the change
> because it's not finished yet.

### PR DESCRIPTION
**Pull request by hagenest**
This PR adds our new OBS repository for our patched dovecot version. Closes #72 and #249.

### CODE REVIEW

File: cmdeploy/src/cmdeploy/__init__.py [hunk 1]
```diff
@@ -522,6 +522,25 @@ def deploy_chatmail(config_path: Path) -> None:
...
+        src = importlib.resources.files(__package__).joinpath("obs-home-deltachat.gpg"),
+        dest = "/etc/apt/keyrings/obs-home-deltachat.gpg",
+        user="root",
+        group="root",
+        mode="644",
+    )
+
+    files.line(
+        name = "Add DeltaChat OBS home repository to sources.list",
+        path = "/etc/apt/sources.list",
```
Comment (missytake) [inline cmdeploy/src/cmdeploy/__init__.py:537]:
```suggestion
        path = "/etc/apt/sources.list.d/opensuse-build-service.list",
```

Let's move this to the subdirectory :) that's what it's good for.

File: cmdeploy/src/cmdeploy/__init__.py [hunk 2]
```diff
@@ -522,6 +522,25 @@ def deploy_chatmail(config_path: Path) -> None:
         packages="postfix",
     )
 
+    # Add our OBS repository for dovecot_no_delay
+    files.put(
+        name = "Add Deltachat OBS GPG key to apt keyring",
+        src = importlib.resources.files(__package__).joinpath("obs-home-deltachat.gpg"),
+        dest = "/etc/apt/keyrings/obs-home-deltachat.gpg",
+        user="root",
+        group="root",
```
Comment (missytake) [inline cmdeploy/src/cmdeploy/__init__.py:531]: I think `user` and `group` can be left out? by default it should be root already, as the root user is used for the deploy.

File: cmdeploy/src/cmdeploy/__init__.py [hunk 3]
```diff
@@ -522,6 +522,25 @@ def deploy_chatmail(config_path: Path) -> None:
...
+        mode="644",
```
Comment (missytake) [inline cmdeploy/src/cmdeploy/__init__.py:532]: Is this necessary? depends on root's umask, I guess. But doesn't hurt.

File: cmdeploy/src/cmdeploy/__init__.py [hunk 4]
```diff
@@ -522,6 +522,25 @@ def deploy_chatmail(config_path: Path) -> None:
...
+    )
+
+    files.line(
+        name = "Add DeltaChat OBS home repository to sources.list",
+        path = "/etc/apt/sources.list",
+        line = "deb [signed-by=/etc/apt/keyrings/obs-home-deltachat.gpg] https://download.opensuse.org/repositories/home:/deltachat/Debian_12/ ./",
+        ensure_newline = True,
+    )
+
+    apt.update(name="apt update")
```
Comment (missytake) [inline cmdeploy/src/cmdeploy/__init__.py:542]: We do an `apt update` in line 486 anyway, maybe leave this out and just move the two other operations above that?

**Review by missytake — APPROVED** [review]
Wow! I just logged into staging.testrun.org, messaged echo, and it came back immediately. Also really nice that we can now use staging.testrun.org to preview PRs.

**Comment by hagenest** [comment]
ARM users take the build for ARM. The OBS documentation was outdated, there was a runner for ARM Debian the entire time

File: cmdeploy/src/cmdeploy/__init__.py (hunk 3 above)
Comment (hagenest) [reply cmdeploy/src/cmdeploy/__init__.py:532]: well, I just copied it from the code below, for consistency :D don't know if it was necessary there

File: cmdeploy/src/cmdeploy/__init__.py (hunk 4 above)
Comment (hagenest) [reply cmdeploy/src/cmdeploy/__init__.py]: done

File: cmdeploy/src/cmdeploy/__init__.py (hunk 3 above)
Comment (missytake) [reply cmdeploy/src/cmdeploy/__init__.py]: Yeah, then let's just merge it and try out in a separate PR if we can leave these out.

File: cmdeploy/src/cmdeploy/__init__.py (hunk 2 above)
Comment (missytake) [reply cmdeploy/src/cmdeploy/__init__.py]: let's just merge it and try out in a separate PR if we can leave these out.

**Comment by hpk42** [comment]
Great but changelpg entry is missing

**Comment by hagenest** [comment]
oops, forgot. pushed a changelog to main, hope that's ok @hpk42

**Pull request merged**